Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -90,16 +231,79 @@ pathManagerState& cruisingState::getInstance()
+    
+    CoordinatedTurnInput_t turnInput {};
+    AltitudeAirspeedInput_t altAirspeedInput {};                                   
+
+    
     //get elevation and turning data
-    if(isError)
+    //loading in commands data from each of the states
+    switch(pathMgr->stage){
+        case ROLL:
+            LandingManager::translateLTSFCommandsToCoordTurns(takeoffRollStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case CLIMB:
+            LandingManager::translateLTSFCommandsToCoordTurns(takeoffClimbStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case TRANSITION:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingTransitionStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case SLOPE:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingSlopeStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case FLARE:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingFlareStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case DECRAB:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingDecrabStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case TOUCHDOWN:
+            LandingManager::translateLTSFCommandsToCoordTurns(landingTouchdownStage::getControlOutput(), sensFusionOutput, imudata, &turnInput, &altAirspeedInput);
+            break;
+        case CRUISING:
+            turnInput.currentHeadingTrack = sensFusionOutput->track; // Gets track;
+            turnInput.desiredHeadingTrack = waypointOutput->desiredTrack;
+            turnInput.accY = imudata->accy;
+            
+            // Set up the compute altitude and airspeed function input
+            altAirspeedInput.currentAltitude = sensFusionOutput->altitude;
+            altAirspeedInput.desiredAltitude = waypointOutput->desiredAltitude;
+            altAirspeedInput.currentAirspeed = sensFusionOutput->airspeed;
+            altAirspeedInput.desiredAirspeed = waypointOutput->desiredAirspeed;   
+            break;
+        default:
+            //should default to cruising
+        
+            turnInput.currentHeadingTrack = sensFusionOutput->track; // Gets track;
+            turnInput.desiredHeadingTrack = waypointOutput->desiredTrack;
+            turnInput.accY = imudata->accy;
+            
+            // Set up the compute altitude and airspeed function input
+            altAirspeedInput.currentAltitude = sensFusionOutput->altitude;
+            altAirspeedInput.desiredAltitude = waypointOutput->desiredAltitude;
+            altAirspeedInput.currentAirspeed = sensFusionOutput->airspeed;
```

Comment:
these statements are actually different since they pull commands from each of the different states. If I were to use if statements, there would be just as many cases.

Reply:
Fair enough. Thanks for clearing it up!